uint8 radius
uint8 circles
uint8 times
---